var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -13,41 +13,83 @@ class Elk(MakefilePackage):
     homepage = 'http://elk.sourceforge.net/'
     url      = 'https://sourceforge.net/projects/elk/files/elk-3.3.17.tgz'
 
-    version('3.3.17', sha256='c9b87ae4ef367ed43afc2d43eb961745668e40670995e8e24c13db41b7e85d73')
-
-    # Elk provides these libraries, but allows you to specify your own
-    variant('blas',   default=True,
-            description='Build with custom BLAS library')
-    variant('lapack', default=True,
-            description='Build with custom LAPACK library')
-    variant('fft',    default=True,
-            description='Build with custom FFT library')
-
-    # Elk does not provide these libraries, but allows you to use them
-    variant('mpi',    default=True,
+    version('7.2.42', sha256='73f03776dbf9b2147bfcc5b7c062af5befa0944608f6fc4b6a1e590615400fc6')
+    version('7.1.14', sha256='7c2ff30f4b1d72d5dc116de9d70761f2c206700c69d85dd82a17a5a6374453d2')
+    version('7.0.12', sha256='9995387c681d0e5a9bd52cb274530b23c0370468b6be86f6c90a6ec445cb8a01')
+
+    # what linear algebra packages to use? the choices are
+    # internal - use internal libraries
+    # generic  - use spack-provided blas and lapack
+    # openblas - use openblas specifically, with special support for multithreading.
+    # mkl - use mkl specifically, with special support for multithreading
+    # should be used with fft=mkl
+    # blis - use internal lapack and blas implementation from blis
+    variant('linalg', default='internal', multi=False,
+            description='Build with custom BLAS library',
+            values=('internal', 'generic', 'openblas', 'mkl', 'blis'))
+    # what FFT package to use? The choices are
+    # internal - use internal library
+    # fftw - fftw3 with special code
+    # mkl  - use mklr with fft code
+    # should be used with linalg=mkls
+    variant('fft',    default='internal', multi=False,
+            description='Build with custom FFT library',
+            values=('internal', 'fftw', 'mkl'))
+    #  check that if fft=mkl then linalg=mkl and vice versa.
+
+    conflicts('linalg=mkl', when='fft=fftw')
+    conflicts('linalg=mkl', when='fft=internal')
+    conflicts('fft=mkl', when='linalg=internal')
+    conflicts('fft=mkl', when='linalg=generic')
+    conflicts('fft=mkl', when='linalg=openblas')
+    conflicts('fft=mkl', when='linalg=blis')
+
+    variant('mpi', default=True,
             description='Enable MPI parallelism')
     variant('openmp', default=True,
             description='Enable OpenMP support')
     variant('libxc',  default=True,
             description='Link to Libxc functional library')
+    variant('w90', default=False,
+            description='wannier90 support, requires wannier90 library')
+
+    depends_on('blas', when='linalg=generic')
+    depends_on('lapack', when='linalg=generic')
+    depends_on('mkl', when='linalg=mkl')
+    depends_on('openblas', when='linalg=openblas')
+    depends_on('blis', when='linalg=blis')
```

Comment:
The comment above says that it uses internal blis, not spack blis

Reply:
It uses spack blis and internal lapack. I'll reword the comment.